### PR TITLE
Build the thing

### PR DESCRIPTION
**Pull request by lennym**
See README for details

### CODE REVIEW

File: test/helpers/index.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+global.sinon = require('sinon');
+
+chai.use(require('sinon-chai'));
+
+global.request = (obj) => {
+  return reqres.req(Object.assign({
+    translate: sinon.spy((a) => Array.isArray(a) ? a[0] : a),
+    sessionModel: new Model()
+  }, obj));
+};
```
Comment (joefitter) [inline test/helpers/index.js:17]: nice!

File: index.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+          fields: fields.map(field => this.getFieldData(field, req)).filter(f => f.value)
+        };
+      })
+      .filter(section => section.fields.length);
+  }
+
+  getSectionSettings(settings) {
+    if (settings.sections) {
+      return settings.sections;
+    } else {
```
Comment (joefitter) [inline index.js]: lint doesn't like this else

File: test/index.js [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+  });
+
+  it('extends hof-controllers.base class', () => {
+    expect(controller).to.be.an.instanceOf(Base);
+  });
+
+  describe('locals', () => {
+
+    beforeEach(() => {
+      sinon.stub(Base.prototype, 'locals').returns({ super: true });
```
Comment (joefitter) [inline test/index.js]: lint doesn't like this bracket padding, however I disagree. IMO we should remove this rule in eslint-config-homeoffice, or just turn it off in .eslintrc for now

Comment (lennym) [reply test/index.js]: I agree. Mostly because I always pad my curlies (not a euphemism!) and the rule annoys me. 

I've fixed it now anyway.

**Comment by lennym** [comment]
Now with 💯% more CI-ness.

**Review by joefitter — APPROVED** [review]

**Comment by joefitter** [comment]
@easternbloc @JoeChapman mind having a look?

**Review by JoeChapman — APPROVED** [review]

**Pull request merged**